Upmerge w17 2020

nvlsianpu opened the pull request:
**not ready for review **

rlubos committed:
[nrf noup] Re-apply downstream changes to mcumgr

As mcumgr is no longer part of the Zephyhr's codebase, and we had it
modified downstream, we need to re-apply those changes in this
repository.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>
(cherry picked from commit c8f675a5fa7f62106b9d913844bd5ed6e16ae69e)
(cherry picked from commit 1cae1d874eb5f349b4b294d72c2f02fa02706f2d)
Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

de-nordic committed:
snapshot: Update mcumgr to commit ae4659b4 from upstream

The commit applies changes that have appeared between
 apache/mynewt-mcumgr:ae4659b445b707409fb17d626c806de3ed284720
and
 zephyrproject-rtos/mcumgr:d4e97cd4fc80ff949415062b1c83fd42929e8fe4

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
snapshot: Update mcumgr to commit 40341abe from the upstream

The commit applies changes that have appeared between the last snapshot
update from the upstream:
 apache/mynewt-mcumgr f3b171f2d6d50b0b696f1307bb18e748727fbc80
and the current top on the upstream:
 apache/mynewt-mcumgr 40341abeeac9c93b4aa59c20ee183376c7920d7e

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
snapshot: Update mcumgr to commit 98e8f320 from upstream

The commit applies changes that have appeared between last snapshot
update from upstream
 apache/mynewt-mcumgr: ae4659b445b707409fb17d626c806de3ed284720
and current top on upstream:
 apache/mynewt-mcumgr: 98e8f320b487a775f49bf554ea29fe8958ab506f

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
snapshot: Update mcumgr to commit f3b171f2 from the upstream

The commit applies changes that have appeared between the last snapshot
update from the upstream
 apache/mynewt-mcumgr: c4919dab456ba383431b1936a6fd7524486f9718
and the current top on the upstream:
 apache/mynewt-mcumgr: f3b171f2d6d50b0b696f1307bb18e748727fbc80

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
snapshot: Update mcumgr to commit c4919dab from the upstream

The commit applies changes that have appeared between the last snapshot
update from the upstream
 apache/mynewt-mcumgr: 98e8f320b487a775f49bf554ea29fe8958ab506f
and the current top on the upstream:
 apache/mynewt-mcumgr: c4919dab456ba383431b1936a6fd7524486f9718

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

nvlsianpu committed:
Revert "[nrf noup] Re-apply downstream changes to mcumgr"

This reverts commit 1cae1d874eb5f349b4b294d72c2f02fa02706f2d.

nvlsianpu committed:
[nrf mergeup] upmerge to mcumgr 1843fdc

MCUMgr SHA: 1843fdce2b008b45f9cc8b38dd9a57a575ab50f6

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

mbolivar-nordic commented on CMakeLists.txt:
```diff
@@ -28,7 +28,7 @@ add_subdirectory(util)
 
 target_link_libraries(MCUMGR INTERFACE
-  zephyr_interface
+  ${IMAGE}zephyr_interface
```
We haven't needed the IMAGE prefix for a while now, since @tejlmand rewrote the multi-image patch set. Should be safe to delete.

tejlmand replied:
wonder for how long time those will pop up.
I caught one like that last week in an very old PR that had been rebased.
This is safe to remove.

nvlsianpu replied:
prefix removed

nvlsianpu committed:
[nrf noup] cmake: remove image target prefix

We haven't needed the IMAGE prefix for a while now,
since the multi-image patch set was rewrote.


Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>
